tv_apps/tv-deck/js/channel_manager.js

```diff
@@ -25,7 +30,305 @@
                 'cardlist-changed', this.updatePinnedChannels.bind(this));
   }
 
-  var proto = {};
+  var proto = Object.create(new evt());
+
+  // Current tuner object playing by TV (we may have another tuner for recording
+  // in the future).
+  Object.defineProperty(proto, 'playingTuner', {
+    get: function() {
+      return this.currentTuners[this.playingTunerId];
+    }
+  });
+
+  // Current source object playing by TV (we may have another source for
+  // recording in the future).
+  Object.defineProperty(proto, 'playingSource', {
+    get: function() {
+      if (!this.playingTuner || !this.playingTuner.sources) {
+        return null;
+      }
+      return this.playingTuner.sources[this.playingSourceType];
+    }
+  });
+
+  // Current channel object playing by TV (we may have another source for
+  // recording in the future).
+  Object.defineProperty(proto, 'playingChannel', {
+    get: function() {
+      var playingSource = this.playingSource;
+      var index = playingSource.channelIndexHash[this.playingChannelNumber];
+      if (!this.playingSource || (index !== 0 && !index)) {
```

Comment:
Since you already use `playingSource` variable, why do you use `this.playingSource` here?